cassandra-4/src/main/java/io/debezium/connector/cassandra/AbstractCassandra4CommitLogParser.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.connector.cassandra;
+
+import java.util.List;
+import java.util.Set;
+
+import org.apache.cassandra.db.commitlog.CommitLogPosition;
+import org.apache.cassandra.db.commitlog.CommitLogReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.DebeziumException;
+import io.debezium.connector.base.ChangeEventQueue;
+import io.debezium.connector.cassandra.exceptions.CassandraConnectorTaskException;
+
+public abstract class AbstractCassandra4CommitLogParser {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractCassandra4CommitLogParser.class);
+
+    private final CommitLogReader commitLogReader;
+    protected final List<ChangeEventQueue<Event>> queues;
+    protected final CommitLogProcessorMetrics metrics;
+    private final Cassandra4CommitLogReadHandlerImpl commitLogReadHandler;
+
+    private final CommitLogTransfer commitLogTransfer;
+    private final Set<String> erroneousCommitLogs;
+    protected boolean completePrematurely = false;
+    protected Cassandra4CommitLogProcessor.LogicalCommitLog commitLog;
+    protected int pollingInterval;
+
+    public AbstractCassandra4CommitLogParser(Cassandra4CommitLogProcessor.LogicalCommitLog commitLog, final List<ChangeEventQueue<Event>> queues,
```

Comment:
Could you please move `LogicalCommitLog` out of `Cassandra4CommitLogProcessor` so it is standalone class in the same package? I find it weird that parser depends on some random stuff in log processor. Same situation with `ProcessingResult`.

Reply:
ProcessingResult might be renamed to `CommitLogProcessingResult`.